### ISSUE TITLE
Feature Request:  Allow Stack children to attach to safe areas

### ISSUE THREAD

**Issue by JohnEstropia** [muukii/MondrianLayout#46]
Currently we have `LayoutContainer` for this, but we cannot use it inside a `ZStackBlock`, for example.
Alternatively, we can have `_ZStackItem` support something like `.alignSelf(.attach(.safeArea(.top)))`

**Comment by muukii** [comment]
Thank you.

We might have something another way to have it because ZStackBlock can be used recursively in Structured layout.
It does not always work well ZStackBlock in a complex node trees.

- Root (has SafeArea)
  - ZStackBlock
    - VStackBlock
      - ZStackBlock -> how attaches elmenent to safe area

**Comment by JohnEstropia** [comment]
I see what you mean. In that case, maybe it would be better to construct the safe area container separately from the parent container?
```swift
safeAreaContainer.mondrian.buildSubviews {
    LayoutContainer(attachedSafeAreaEdges: .all) {
        // ...
    }
}
backgroundView.mondrian.buildSubviews {
    ZStackBlock {
        safeAreaContainer
        // ...
    }
}
```

**Comment by muukii** [comment]
@JohnEstropia 

does your example code help your case?

I thought it does not help because that safeAreaContainer would have a safe-area layout guide but that would be related to the parent container. What if we add padding(200) in ZStackBlock, the safeAreaContainer has no matter.

So If I could, I'd like to get something figure to explain what you expect layout.

**Comment by muukii** [comment]
a case

```
      mondrianBatchLayout {
        collectionNode.view.mondrian.layout.edges(.toSuperview)
        overlayViewLayoutGuide.mondrian.layout
          .top(.to(view.safeAreaLayoutGuide))
          .horizontal(.toSuperview)
          .bottom(.toSuperview)
      }
```

**Comment by JohnEstropia** [comment]
Yes, I think it would be better to construct the layout separately (classic layout) for the use-case I had in mind

**Comment by muukii** [comment]
do you think it's better if can we describe multiple LayoutContainer in building subviews?
